ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/BridgeWriteService.scala

```diff
@@ -128,54 +120,23 @@ case class ReadWriteServiceBridge(
       PruningResult.ParticipantPruned
     )
 
-  override def ledgerInitialConditions(): Source[LedgerInitialConditions, NotUsed] =
-    Source.single(
-      LedgerInitialConditions(
-        ledgerId = ledgerId,
-        config = Configuration(
-          generation = 1L,
-          timeModel = LedgerTimeModel.reasonableDefault,
-          maxDeduplicationTime = java.time.Duration.ofSeconds(maxDedupSeconds.toLong),
-        ),
-        initialRecordTime = Timestamp.now(),
-      )
-    )
+  val queue: BoundedSourceQueue[Submission] = {
+    val (queue, queueSource) =
+      Source
+        .queue[Submission](submissionBufferSize)
+        .zipWithIndex
+        .map { case (submission, index) =>
+          (toOffset(index), successMapper(submission, index, participantId))
+        }
+        .preMaterialize()
 
-  var stateUpdatesWasCalledAlready = false
-  override def stateUpdates(
-      beginAfter: Option[Offset]
-  )(implicit loggingContext: LoggingContext): Source[(Offset, Update), NotUsed] = {
-    // TODO for PoC purposes:
-    //   This method may only be called once, either with `beginAfter` set or unset.
-    //   A second call will result in an error unless the server is restarted.
-    //   Bootstrapping the bridge from indexer persistence is supported.
-    synchronized {
-      if (stateUpdatesWasCalledAlready)
-        throw new IllegalStateException("not allowed to call this twice")
-      else stateUpdatesWasCalledAlready = true
-    }
-    logger.info("Indexer subscribed to state updates.")
-    beginAfter.foreach(offset =>
-      logger.warn(
-        s"Indexer subscribed from a specific offset $offset. This offset is not taking into consideration, and does not change the behavior of the ReadWriteServiceBridge. Only valid use case supported: service starting from an already ingested database, and indexer subscribes from exactly the ledger-end."
-      )
+    queueSource.runWith(Sink.fromSubscriber(readServiceWithFeedSubscriber.subscriber))
```

Comment:
Is there absolutely no asynchronicity with this setup ? 

Usually if the same contract has two consuming exercises performed on it in parallel then the both will result in SubmittedTransaction being passed to the write service, only one should succeed.

This appears to assume every SubmittedTransaction will succeed.

Reply:
This is the status quo of Sandbox-on-X: It currently does not perform any conflict checking. We are implementing conflict checking as part of the current work stream.